doc/user/installing-llvm.md

```diff
@@ -92,6 +92,8 @@ We would recommend that you install LLVM 4 via [Homebrew](https://brew.sh).
 $ brew install llvm@4
 ```
 
+You may need to follow the output of `brew install` and add llvm@4 to your `PATH` to make `opt` and `llvm-link` discoverable.
```

Comment:
Thank you for the PR. It shouldn't be necessary to add llvm@4/bin to PATH though when using TruffleRuby releases, as we look for that path when looking for `opt` and `llvm-link` in TruffleRuby.
Could you describe a situation in which it's necessary?

Or maybe did you need this to build TruffleRuby from source?

See also https://github.com/oracle/truffleruby/issues/1583 where we find out where Homebrew is installed.

Reply:
Here's the output from 

```
llvm@4: stable 4.0.1 (bottled) [keg-only]
Next-gen compiler infrastructure
https://llvm.org/
/usr/local/Cellar/llvm@4/4.0.1_1 (4,452 files, 2.0GB)
  Poured from bottle on 2019-07-10 at 09:29:33
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/llvm@4.rb
==> Dependencies
Build: cmake ✘
Required: libffi ✔
==> Caveats
To use the bundled libc++ please add the following LDFLAGS:
  LDFLAGS="-L/usr/local/opt/llvm@4/lib -Wl,-rpath,/usr/local/opt/llvm@4/lib"

llvm@4 is keg-only, which means it was not symlinked into /usr/local,
because this is an alternate version of another formula.

If you need to have llvm@4 first in your PATH run:
  echo 'export PATH="/usr/local/opt/llvm@4/bin:$PATH"' >> ~/.zshrc

For compilers to find llvm@4 you may need to set:
  export LDFLAGS="-L/usr/local/opt/llvm@4/lib"
  export CPPFLAGS="-I/usr/local/opt/llvm@4/include"
```

Before I've used `export PATH="/usr/local/opt/llvm@4/bin:$PATH"`, `jt build` was failing with:

```
$ /Users/kir/src/github.com/oracle/mx/mx --java-home /Users/kir/src/github.com/oracle/openjdk1.8.0_212-jvmci-20-b04/Contents/Home -p /Users/kir/src/github.com/oracle/truffleruby --dynamicimports /vm build
JAVA_HOME: /Users/kir/src/github.com/oracle/openjdk1.8.0_212-jvmci-20-b04/Contents/Home
EXTRA_JAVA_HOMES:
Dependencies removed from build:
 project com.oracle.truffle.dsl.processor.jdk9 was removed as JDK 9 is not available
Non-default dependencies removed from build (use mx build --all to build them):
 com.oracle.truffle.llvm.tests.interop
 com.oracle.truffle.llvm.tests.irdebug
 parserTorture
 gcc_fortran
 com.oracle.truffle.llvm.tests.sulongavx
 com.oracle.truffle.llvm.pipe.native
 com.oracle.truffle.llvm.test.native
 com.oracle.truffle.llvm.tests.bitcodeformat
 gcc_c
 com.oracle.truffle.llvm.pipe
 com.oracle.truffle.llvm.tests.sulongcpp
 nwcc
 inlineassemblytests
 com.oracle.truffle.llvm.tests.sulong
 com.oracle.truffle.llvm.tests.libc
 other
 com.oracle.truffle.llvm.tests.other
 com.oracle.truffle.llvm.tests.debug
 llvm
 com.oracle.truffle.llvm.tests.bitcode
 shootout
 gcc_cpp
 JACOCOREPORT_0.8.4
 SULONG_TEST_NATIVE
 SULONG_TEST
 SULONG_TEST_SUITES
Compiling com.oracle.mxtool.compilerserver with javac(JDK 1.8)... [/Users/kir/src/github.com/oracle/mx/mxbuild/java/com.oracle.mxtool.compilerserver/bin/com/oracle/mxtool/compilerserver/ECJDaemon.class does not exist]
Compiling org.graalvm.options with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/graal/sdk/mxbuild/src/org.graalvm.options/bin/org/graalvm/options/OptionValues.class does not exist]
Compiling org.graalvm.word with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/graal/sdk/mxbuild/src/org.graalvm.word/bin/org/graalvm/word/Pointer.class does not exist]
Compiling org.truffleruby.annotations with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/truffleruby/mxbuild/org.truffleruby.annotations/bin/org/truffleruby/PopulateBuildInformation.class does not exist]
Compiling org.graalvm.collections with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/graal/sdk/mxbuild/src/org.graalvm.collections/bin/org/graalvm/collections/UnmodifiableEconomicMap.class does not exist]
Compiling com.oracle.mxtool.checkcopy with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/mx/mxbuild/java/com.oracle.mxtool.checkcopy/bin/com/oracle/mxtool/checkcopy/CheckCopyright.class does not exist]
Compiling org.truffleruby.ruby with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/truffleruby/mxbuild/org.truffleruby.ruby/bin/truffleruby/core/complex.rb does not exist]
Building com.oracle.truffle.llvm.libraries.bitcode with GNU Make... [rebuild needed by GNU Make]
Compiling com.oracle.truffle.llvm.pipe with javac-daemon(JDK 1.8)... [/Users/kir/src/github.com/oracle/graal/sulong/mxbuild/projects/com.oracle.truffle.llvm.pipe/bin/com/oracle/truffle/llvm/pipe/CaptureOutput.class does not exist]
Archiving CHROMEINSPECTOR_GRAALVM_SUPPORT... [/Users/kir/src/github.com/oracle/graal/tools/mxbuild/dists/chromeinspector-graalvm-support.tar does not exist]
Archiving SULONG_GRAALVM_DOCS... [/Users/kir/src/github.com/oracle/graal/sulong/mxbuild/darwin-amd64/dists/sulong-graalvm-docs.tar does not exist]
Archiving TRUFFLE_PROFILER_GRAALVM_SUPPORT... [/Users/kir/src/github.com/oracle/graal/tools/mxbuild/dists/truffle-profiler-graalvm-support.tar does not exist]
Archiving TRUFFLERUBY_GRAALVM_LICENSES... [/Users/kir/src/github.com/oracle/truffleruby/mxbuild/darwin-amd64/dists/truffleruby-graalvm-licenses.tar does not exist]
Archiving TRUFFLE_NFI_GRAALVM_SUPPORT... [/Users/kir/src/github.com/oracle/graal/truffle/mxbuild/dists/truffle-nfi-graalvm-support.tar does not exist]
Archiving TRUFFLE_GRAALVM_SUPPORT... [/Users/kir/src/github.com/oracle/graal/truffle/mxbuild/dists/truffle-graalvm-support.tar does not exist]
Archiving WORD_API... [dependency org.graalvm.word updated]

make: opt: No such file or directory
make: *** [bin/abort.bc] Error 1
rm bin/abort.noopt.bc
```

Reply:
Right, that makes sense :)
What fails here is actually building Sulong, which requires `opt` in `PATH`.
It's documented in https://github.com/oracle/graal/tree/master/sulong#macos but we don't link there in `workflow.md` (which are instructions specifically to build TruffleRuby vs `doc/user/installing-llvm.md` which is meant for users using pre-built releases).

Therefore, could you add this documentation to https://github.com/oracle/truffleruby/blob/master/doc/contributor/workflow.md#requirements under `Requirements` and link to the Sulong README ?

Reply:
Thanks, updated!